File: src/ui/Deck.js
import Api from "./Api.js";
import Hand from "./Hand.js";
import { useState, useCallback } from "react";

export default class Deck {
  static async create() {
    const ID = await Api.deck(); // create the deck on the server and get the id
    return new Deck(ID.message);
  }

  constructor(ID) {
    this.deckID = ID;
  }
  //returns the deck ID
  async getDeck() {
    return this.deckID;
  }
  //deals count number of cards
  async deal(count) {
    return await Api.dealV2(this.deckID, count);
  }
  //converts a hand(an array of card objects) to a string in order to pass cards/hands to the backend
  async strRepCards(hand, selected) {
    if (selected.length === 0) {
      return "NONE";
    }
    let cardsString = "";
    for (let i = 0; i < selected.length; i++) {
      cardsString += hand[selected[i]].rank;
      cardsString += hand[selected[i]].suit;
    }
    return cardsString;
  }

  async highestRank(hand) {
    const cards = await this.strRepCards(hand, [0, 1, 2, 3, 4]);

    return await Api.highestRank(cards);
  }

  // sends cards to return to the backend deck
  async discard(hand, selected, newGame) {
    if (newGame) {
      const cards = await this.strRepCards(hand, [0, 1, 2, 3, 4]);
      return await Api.discard(this.deckID, cards);
    }

    const cards = await this.strRepCards(hand, selected);
    return await Api.discard(this.deckID, cards);
  }
}

File: src/deck.py
from card import Card
import random

SUITS = ["C", "D", "H", "S"]
RANKS = ["2", "3", "4", "5", "6", "7", "8", "9", "0", "J", "Q", "K", "A"]


class Deck:
    
    def __init__(self, id):
        print("Deck constructor")
        self.deck = [Card(rank, suit) for rank in RANKS for suit in SUITS]
        self.shuffle()
        self.id = id

    def drawCard(self):
        return self.deck.pop(0)

    def discard(self, card):
        self.deck.append(card)

    
    def shuffle(self):
        random.shuffle(self.deck)